ngraph/frontend/onnx_import/src/op/random_uniform.cpp

```diff
@@ -0,0 +1,102 @@
+                                                      bool>::type = true>
+                    OutputVector
+                        gen_random_constant(const Shape& shape, float low, float high, float seed)
+                    {
+                        std::mt19937 gen(seed);
+                        std::uniform_real_distribution<T> dis(low, high);
+                        size_t n = shape_size(shape);
+                        std::vector<T> values;
+                        values.reserve(n);
+                        for (size_t i = 0; i < n; i++)
+                            values.push_back(dis(gen));
+                        return {default_opset::Constant::create(element::from<T>(), shape, values)};
+                    }
+
+                    template <
+                        typename T,
+                        typename std::enable_if<std::is_integral<T>::value, bool>::type = true>
+                    OutputVector
+                        gen_random_constant(const Shape& shape, float low, float high, float seed)
+                    {
+                        std::mt19937 gen(seed);
+                        std::uniform_int_distribution<T> dis(low, high);
+                        size_t n = shape_size(shape);
+                        std::vector<T> values;
+                        values.reserve(n);
+                        for (size_t i = 0; i < n; i++)
+                            values.push_back(dis(gen));
+                        return {default_opset::Constant::create(element::from<T>(), shape, values)};
+                    }
+                } // namespace
+            }     // namespace detail
+            namespace set_1
+            {
+                OutputVector random_uniform(const Node& node)
+                {
+                    float low = node.get_attribute_value<float>("low", 0.0f);
+                    float high = node.get_attribute_value<float>("high", 1.0f);
+                    float seed = node.get_attribute_value<float>("seed", std::random_device()());
+
+                    auto shape_attr = node.get_attribute_value<std::vector<int64_t>>("shape");
+                    Shape shape;
+                    std::transform(shape_attr.begin(),
+                                   shape_attr.end(),
+                                   std::back_inserter(shape),
+                                   [](int64_t dim) -> size_t { return static_cast<size_t>(dim); });
+
+                    using ONNX_NAMESPACE::TensorProto;
+                    TensorProto::DataType dtype =
+                        static_cast<TensorProto::DataType>(node.get_attribute_value<int64_t>(
+                            "dtype", static_cast<int64_t>(TensorProto::FLOAT)));
+                    switch (dtype)
+                    {
+                    case TensorProto::FLOAT:
+                        return detail::gen_random_constant<float>(shape, low, high, seed);
```

Comment:
Is returning a constant a correct behavior here?
It have sense only if seed is pre-definied in attributes IMO.

Reply:
Can you elaborate more? Why is it not okay to return constant if seed attribute is not provided and we have to generate one?

Reply:
On a second thought I think that even if seed is pre-definied we should not return a constant value.
This approach works only if a model is run only once after import.
But if the model is imported and then several iterations are run we won't have random values between iterations.
Current solution is similar to roll a dice and hardcode a score in code.
I don't think it is a proper way to implement this operator.

Reply:
Yes, you're right.